GooglePlayInstant/Editor/QuickDeploy/OAuth2Server.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using UnityEngine;
+
+[assembly: InternalsVisibleTo("GooglePlayInstant.Tests.Editor.QuickDeploy")]
+
+namespace GooglePlayInstant.Editor.QuickDeploy
+{
+    /// <summary>
+    /// A class representing a server that provides an endpoint to use for getting authorization code from Google's
+    /// OAuth2 API's authorization page. On that page, the user grants the application to access their data on Google
+    /// Cloud platform. The server attempt to open and start listening at a fixed port. The server announces
+    /// its chosen endpoint with the CallbackEndpoint property. This server will run until it receives the first
+    /// request, which it will process to retrieve the authorization response and handle the by invoking on the response
+    /// the handler passed to the server during instatiation. The server will then stop listening for further requests.
+    ///
+    /// <see cref="https://developers.google.com/identity/protocols/OAuth2#installed"/> for an overview of OAuth2
+    /// protocol for installed applications that interact with Google APIs.
+    /// </summary>
+    public class OAuth2Server
+    {
+        private const string CloseTabText = "You may close this tab.";
```

Comment:
Add a comment:

// Arbitrarily chosen port to listen for the authorization callback.

Reply:
Resolved.